src/main/java/com/minecolonies/tileentities/ScarecrowTileEntity.java

```diff
@@ -0,0 +1,91 @@
+package com.minecolonies.tileentities;
+
+import java.util.Random;
+import com.minecolonies.inventory.InventoryField;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+
+/**
+ * The scarecrow tile entity to store extra data.
+ */
+public class ScarecrowTileEntity extends TileEntity
+{
+    /**
+     * NBTTag to store the type.
+     */
+    private static final String TAG_TYPE = "type";
+
+    /**
+     * The name of the inventory which shows up when opened.
+     */
+    private static final String INVENTORY_NAME = "Scarecrow";
+    //todo change to .lang term
+
+    /**
+     * The inventory connected with the scarecrow.
+     */
+    public InventoryField inventoryField;
+
+    /**
+     * Random generator.
+     */
+    private final Random random = new Random();
+
+    /**
+     * The type of the scarecrow.
+     */
+    private ScareCrowType TYPE;
+
+    /**
+     * Enum describing the different textures the scarecrow has.
+     */
+    public enum ScareCrowType
+    {
+        PUMPKINHEAD,
+        NORMAL
+    }
+
+    /**
+     * Empty public constructor.
+     */
+    public ScarecrowTileEntity()
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) It is a good practice to call super() in a constructor [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ACallSuperInConstructor)